File: backend/src/main/java/com/realdolmen/backend/domain/Game.java
package com.realdolmen.backend.domain;

import javax.persistence.*;
import javax.validation.constraints.Size;

@Entity
@DiscriminatorValue("GAME")
public class Game extends Article {
    @Size(max = 100)
    private String publisher;

    @Column(name = "minimum_age")
    private int minimumAge;

    @Enumerated(EnumType.STRING)
    private GameGenre genre;

    public String getPublisher() {
        return publisher;
    }

    public void setPublisher(String publisher) {
        this.publisher = publisher;
    }

    public int getMinimumAge() {
        return minimumAge;
    }

    public void setMinimumAge(int minimumAge) {
        this.minimumAge = minimumAge;
    }

    public GameGenre getGenre() {
        return genre;
    }

    public void setGenre(GameGenre genre) {
        this.genre = genre;
    }

    @Override
    public String getDisplayValue(){
        return "Game";
    }
}

File: frontend/angular/src/app/entity/Game.ts
import {Article} from "./Article";

export class Game extends Article{
  private _genre: string;
  private _publisher: string;
  private _minimumAge: number;

  constructor() {
    super();
  }

  get genre(): string {
    return this._genre;
  }

  set genre(value: string) {
    this._genre = value;
  }

  get publisher(): string {
    return this._publisher;
  }

  set publisher(value: string) {
    this._publisher = value;
  }

  get minimumAge(): number {
    return this._minimumAge;
  }

  set minimumAge(value: number) {
    this._minimumAge = value;
  }
}
